bawan-shna-abdulrahman/src/Components/Search.js

```diff
@@ -0,0 +1,69 @@
+import React, { useState, useEffect } from "react";
+import { Button, Form, FormControl } from "react-bootstrap";
+import Spinners from "./Spinner";
+import DropdownCategories from "./DropdownCategories";
+import { constructUrl } from "./Api";
+
+export default function Search(props) {
+  const [category, setCategory] = useState({});
+
+  const changeCategory = (category) => {
+    props.setIsLoading(true);
+    setCategory(category);
+  };
+  const [query, setQuery] = useState("");
+  const onChange = (e) => {
+    setQuery(e.target.value);
+  };
+  const onSubmit = (event) => {
+    event.preventDefault();
+    props.setIsLoading(true);
+    props.handleQuery(query);
+  };
+
+  useEffect(fetchMovies, [props.isLoading, category]);
+  function fetchMovies() {
+    if (!props.isLoading) return;
+    let SEARCH_URL;
+    if (query !== "") {
+      SEARCH_URL = constructUrl("search/movie", query);
+    } else {
+      SEARCH_URL = constructUrl("movie/popular");
+    }
+    fetch(SEARCH_URL)
+      .then((res) => res.json())
+      .then((data) => {
+        console.log(data);
+        if (data.results !== undefined) {
+          let movies = data.results;
```

Comment:
when ever you can use const and without no then problems use it.

Reply:
if i use const here when you change the categories it will not fetch the movies of that category